pkg/collector/corechecks/ebpf/c/oom-kill-kern.c

```diff
@@ -0,0 +1,70 @@
+#include <linux/compiler.h>
+#include <linux/kconfig.h>
+#include <linux/ptrace.h>
+#include <linux/types.h>
+#include <linux/version.h>
+#include <linux/oom.h>
+
+#include "bpf_helpers.h"
+#include "bpf-common.h"
+#include "oom-kill-kern-user.h"
+
+
+/*
+ * The `oom_stats` hash map is used to share with the userland program system-probe
+ * the statistics per pid
+ */
+
+struct bpf_map_def SEC("maps/oom_stats") oom_stats = {
+    .type = BPF_MAP_TYPE_HASH,
+    .key_size = sizeof(u32),
+    .value_size = sizeof(struct oom_stats),
+    .max_entries = 10240,
+    .pinning = 0,
+    .namespace = "",
+};
+
+SEC("kprobe/oom_kill_process")
+int kprobe__oom_kill_process(struct pt_regs *ctx) {
+    struct oom_control *oc;
+    bpf_probe_read(&oc, sizeof(oc), &PT_REGS_PARM1(ctx));
+
+    struct oom_stats zero = {};
+    u32 pid = bpf_get_current_pid_tgid() >> 32;
+
+    struct oom_stats* s = bpf_map_lookup_elem(&oom_stats, &pid);
+    if (s == NULL) {
+        if (bpf_map_update_elem(&oom_stats, &pid, &zero, BPF_NOEXIST))
```

Comment:
If you want to avoid a race condition, the order should `bpf_map_update_elem` with `BPF_NOEXIST` then `bpf_map_lookup_elem`. See https://github.com/DataDog/datadog-agent/blob/master/pkg/network/ebpf/c/tracer-stats.h#L34-L37

Reply:
:+1: